lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,353 @@
+  int _errorCount = 0;
+
+  /// Map containing Xunit hierarchy.
+  Map _groupStructure = {};
+
+  /// Watches the tests run by [engine] and records its results.
+  static XunitReporter watch(Engine engine, {bool verboseTrace: false}) {
+    return new XunitReporter._(engine, verboseTrace: verboseTrace);
+  }
+
+  XunitReporter._(this._engine, {bool verboseTrace: false}) {
+    _subscriptions.add(_engine.onTestStarted.listen(_onTestStarted));
+
+    /// Convert the future to a stream so that the subscription can be paused or
+    /// canceled.
+    _subscriptions.add(_engine.success.asStream().listen(_onDone));
+  }
+
+  void pause() {
+    if (_paused) return;
+    _paused = true;
+
+    _stopwatch.stop();
+
+    for (var subscription in _subscriptions) {
+      subscription.pause();
+    }
+  }
+
+  void resume() {
+    if (!_paused) return;
+    if (_stopwatchStarted) _stopwatch.start();
+
+    for (var subscription in _subscriptions) {
+      subscription.resume();
+    }
+  }
+
+  void cancel() {
+    for (var subscription in _subscriptions) {
+      subscription.cancel();
+    }
+    _subscriptions.clear();
+  }
+
+  /// A callback called when the engine begins running [liveTest].
+  void _onTestStarted(LiveTest liveTest) {
+    if (liveTest.suite is! LoadSuite) {
+      if (!_stopwatch.isRunning) _stopwatch.start();
+
+      _subscriptions.add(liveTest.onStateChange
+          .listen((state) => _onStateChange(liveTest, state)));
+    }
+
+    _subscriptions.add(liveTest.onError
+        .listen((error) => _onError(liveTest, error.error, error.stackTrace)));
+
+    _subscriptions.add(liveTest.onPrint.listen((line) {}));
+  }
+
+  /// Return an ordered list of groups that [liveTest] belongs to
+  List _orderedGroupList(List<Group> listOfGroups) {
+    List separatedList = [];
+    if (listOfGroups.length > 1) {
+      separatedList.add(listOfGroups[1].name);
+      for (int i = 2; i < listOfGroups.length; i++) {
+        separatedList.add(listOfGroups[i]
+            .name
+            .replaceAll(listOfGroups[i - 1].name, '')
+            .trim());
+      }
+    }
+    return separatedList;
+  }
+
+  /// A callback called when [liveTest]'s state becomes [state].
+  void _onStateChange(LiveTest liveTest, State state) {
+    if (state.status != Status.complete) {
+      if (state.status == Status.running &&
+          !_testCases.containsKey(liveTest.test)) {
+        String testName = liveTest.individualName;
+        testName = testName.replaceAll('&', '&amp;');
+        testName = testName.replaceAll('<', '&lt;');
+        testName = testName.replaceAll('>', '&gt;');
+        testName = testName.replaceAll('"', '&quot;');
+        testName = testName.replaceAll("'", '&apos;');
+
+        _testCases[liveTest] = new XunitTestResult(
+            testName, liveTest.suite.path, _stopwatch.elapsedMilliseconds);
+        List listOfGroups = _orderedGroupList(liveTest.groups);
+        Map listGroup = _groupStructure;
```

Comment:
This assignment seems unnecessary, could you just use `_groupStructure` directly?